runtime/src/accounts_index.rs

```diff
@@ -1205,7 +1209,7 @@ impl<T: 'static + Clone + IsCached + ZeroLamport> AccountsIndex<T> {
         &self,
         slot: Slot,
         items: Vec<(&Pubkey, T)>,
-    ) {
+    ) -> Vec<bool> {
```

Comment:
This is a good idea. I've been working towards better leveraging the repeated scans over all storages at startup. I had imagined creating the index or calculating the hash could return by-products of useful info for clean or shrink.